### ISSUE TITLE
SNS: unable to ConfirmSubscription: Topic not found

### ISSUE THREAD

**Issue by NoumanSaleem** [localstack/localstack#1808]
Hi all,

Thanks for your effort on localstack! I'm trying to locally test SNS (HTTP) w/ cloudwatch triggers, but am unable to get past confirming the subscription.

My application receives the following POST body when creating a subscription:
```
{"MessageId": "5cb062ad-0d4e-41e6-9a80-7053926b20b4", "Type": "SubscriptionConfirmation", "Timestamp": "2019-11-27T04:29:21.166530Z", "Message": "You have chosen to subscribe to the topic arn:aws:sns:us-e
ast-1:000000000000:lambda-xyz-errors.\nTo confirm the subscription, visit the SubscribeURL included in this message.", "TopicArn": "arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors", "Token": "3f97b1$
2", "SubscribeURL": "http://b40035e82fc6:4575/?Action=ConfirmSubscription&TopicArn=arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors&Token=3f97b192"}
```

If I curl the SubscribeURL, I get a `topic does not exist` error:

```
[I] ➜ curl -XGET -H 'Authorization: AWS4-HMAC-SHA256 Credential=AKIAIOSFODNN7EXAMPLE/20130524/us-east-1/sns/aws4_request,SignedHeaders=host;range;x-amz-date,Signature=fe5f80f77d5fa3beca038a248ff027d044534
2fe2855ddc963176630326f1024' http://localhost:4575/\?Action\=ConfirmSubscription\&TopicArn\=arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors\&Token\=75f32aec
<ErrorResponse xmlns="http://sns.amazonaws.com/doc/2010-03-31/">
  <Error>
    <Type>Sender</Type>
    <Code>NotFound</Code>
    <Message>Topic does not exist</Message>
  </Error>
  <RequestId>9dd01905-5012-5f99-8663-4b3ecd0dfaef</RequestId>
</ErrorResponse>%
```

If I run list-topics against the container, I can see it exists:

```bash
docker-compose exec localstack awslocal sns list-topics
{
    "Topics": [
        {
            "TopicArn": "arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors"
        }
    ]
}
```

The topic and subscription were created with:
```bash
awslocal sns create-topic --name lambda-xyz-errors
awslocal sns subscribe --topic-arn arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors --protocol http --notification-endpoint "http://localhost:3000/"

```

**Comment by whummer** [comment]
Thanks for reporting @NoumanSaleem . Can you please try starting LocalStack with the `LOCALSTACK_HOSTNAME` configuration:
```
LOCALSTACK_HOSTNAME=localhost SERVICES=sns localstack start
```
If the problem persists, please report the full `docker run ...` command that gets printed on the terminal when you start up LocalStack. Thanks

**Comment by NoumanSaleem** [comment]
Hey @whummer, I get the same outcome running with `LOCALSTACK_HOSTNAME=localhost`.

```
[I] ➜ curl -XGET -H 'Authorization: AWS4-HMAC-SHA256 Credential=AKIAIOSFODNN7EXAMPLE/20130524/us-east-1/sns/aws4_request,SignedHeaders=host;range;x-amz-date,Signature=fe5f80f77d5fa3beca038a248ff027d044534
2fe2855ddc963176630326f1024' 'http://localhost:4575/?Action=ConfirmSubscription&TopicArn=arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors&Token=87881978'
<ErrorResponse xmlns="http://sns.amazonaws.com/doc/2010-03-31/">
  <Error>
    <Type>Sender</Type>
    <Code>NotFound</Code>
    <Message>Topic does not exist</Message>
  </Error>
  <RequestId>9dd01905-5012-5f99-8663-4b3ecd0dfaef</RequestId>
</ErrorResponse>%
```

```
docker-compose exec localstack awslocal sns list-topics
{
    "Topics": [
        {
            "TopicArn": "arn:aws:sns:us-east-1:000000000000:lambda-xyz-errors"
        }
    ]
}
```

Here is my dockerfile:

```
version: '2.1'
services:
  localstack:
    image: localstack/localstack
    ports:
      - "4567-4593:4567-4593"
      # - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES-iam,cloudwatch,cloudwatchlogs,sns,lambda}
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR-/tmp/localstack}
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR-docker}
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - HOSTNAME_EXTERNAL=localhost
      - DOCKER_HOST=unix:///var/run/docker.sock
      - LOCALSTACK_HOSTNAME=localhost
    volumes:
      - "/private/${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"

```

**Comment by NoumanSaleem** [comment]
Should we expect to see `req_action == 'ConfirmSubscription'` somewhere in  https://github.com/localstack/localstack/blob/master/localstack/services/sns/sns_listener.py ?

**Comment by whummer** [comment]
Looks like this part is currently still missing - do you think you could take a stab at it an contribute a pull request to fix this issue @NoumanSaleem ? That would be really appreciated, thanks!

**Comment by NoumanSaleem** [comment]
@whummer hope I can peel away some time to try it!

**Comment by whummer** [comment]
This should be fixed in #2043 - please give it a try with the latest version and let us know if the problem persists @NoumanSaleem . Thanks
